src/components/Form/FormDatePicker.react.js

```diff
@@ -0,0 +1,173 @@
+// @flow
+
+import * as React from "react";
+import FormSelect from "./FormSelect.react";
+import FormInputGroup from "./FormInputGroup.react";
+
+type Props = {|
+  +children?: React.Node,
+  +className?: string,
+  +defaultDate: Date,
+  +minYear: number,
+  +maxYear: number,
+  +format: string,
+  +monthLabels: Array<string>,
+  +onChange?: Date => void,
+|};
+
+type State = {|
+  currentDate: Date,
+|};
+
+type DateComponents = {
+  yyyy: React.Node,
+  dd: React.Node,
+  mm: React.Node,
+};
+
+type ChangeTypes = "mm" | "yyyy" | "dd";
+
+class FormDatePicker extends React.PureComponent<Props, State> {
+  static defaultProps = {
+    monthLabels: [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ],
+    minYear: 1897,
+    maxYear: new Date().getFullYear(),
+    format: "mm/dd/yyyy",
+    defaultDate: new Date(),
+  };
+
+  // Set the default date from props
+  componentWillMount(): void {
```

Comment:
Sorry to be annoying about this - I'd rather avoid using this lifecycle method and simply use the state constructor instead:

```jsx
state = {
    currentDate: this.props.defaultDate,
};
```

Reply:
@AaronCoplan I didn't know we could do that lol. Makes perfect sense, will do that as soon as I leave work